tests/tests/test-crowdloan.ts

```diff
@@ -0,0 +1,434 @@
+    sudoAccount = await keyring.addFromUri(ALITH_PRIV_KEY, null, "ethereum");
+  });
+  it("should be able to make a first claim", async function () {
+    await context.polkadotApi.tx.sudo
+      .sudo(
+        context.polkadotApi.tx.crowdloanRewards.initializeRewardVec(
+          [[relayChainAddress, GENESIS_ACCOUNT, 3_000_000n * GLMR]],
+          0,
+          1
+        )
+      )
+      .signAndSend(sudoAccount);
+    await context.createBlock();
+
+    let reward_info = (
+      await context.polkadotApi.query.crowdloanRewards.accountsPayable(GENESIS_ACCOUNT)
+    ).toJSON() as any;
+    let claimed = await calculate_vested_amount(
+      context,
+      reward_info.total_reward,
+      reward_info.claimed_reward,
+      1
+    );
+    // construct a transaction
+    const transfer = context.polkadotApi.tx.crowdloanRewards.claim();
+    await transfer.signAndSend(genesisAccount);
+
+    await context.createBlock();
+
+    expect(
+      (
+        (
+          await context.polkadotApi.query.crowdloanRewards.accountsPayable(GENESIS_ACCOUNT)
+        ).toHuman() as any
+      ).claimed_reward
+    ).to.equal(claimed);
+
+    // check balances
+    expect(
+      (
+        Number(await context.web3.eth.getBalance(GENESIS_ACCOUNT)) - Number(GENESIS_ACCOUNT_BALANCE)
+      ).toString()
+    ).to.equal("6.000238095238095e+23");
+    const account = await context.polkadotApi.query.system.account(GENESIS_ACCOUNT);
+    expect(
+      (Number(account.data.free.toString()) - Number(GENESIS_ACCOUNT_BALANCE)).toString()
+    ).to.equal("6.000238095238095e+23");
+  });
+});
+
+describeDevMoonbeam("Crowdloan", (context) => {
+  let genesisAccount: KeyringPair, sudoAccount: KeyringPair;
+  const relayChainAddress: string = "couldBeAnyString";
+  before("Setup genesis account for substrate", async () => {
+    const keyring = new Keyring({ type: "ethereum" });
+    genesisAccount = await keyring.addFromUri(GENESIS_ACCOUNT_PRIVATE_KEY, null, "ethereum");
+    sudoAccount = await keyring.addFromUri(ALITH_PRIV_KEY, null, "ethereum");
+  });
+  it("should show me the money after 5 blocks, after first claim was called", async function () {
+    await context.polkadotApi.tx.sudo
+      .sudo(
+        context.polkadotApi.tx.crowdloanRewards.initializeRewardVec(
+          [[relayChainAddress, GENESIS_ACCOUNT, 3_000_000n * GLMR]],
+          0,
+          1
+        )
+      )
+      .signAndSend(sudoAccount);
+    await context.createBlock();
+
+    let reward_info = (
+      await context.polkadotApi.query.crowdloanRewards.accountsPayable(GENESIS_ACCOUNT)
+    ).toJSON() as any;
+
+    // construct a transaction
+    const transfer = context.polkadotApi.tx.crowdloanRewards.claim();
+    await transfer.signAndSend(genesisAccount);
+
+    await context.createBlock();
+
+    // should show me the money after 5 blocks
+    // only works with a relaychain
+    await context.createBlock();
+    await context.createBlock();
+    await context.createBlock();
```

Comment:
The comment is outdated now right?
Would be nice to make the block creation into a for loop or a helper function or something.

Reply:
@JoshOrndorff removed